Makefile

```diff
@@ -1,118 +1,35 @@
-
-.PHONY: all test clean images protos assets check_assets release-prepare release-perform
 .DEFAULT_GOAL := all
+.PHONY := all images check-generated-files logcli loki loki-debug promtail promtail-debug lint test clean yacc protos helm helm-install helm-upgrade helm-publish helm-debug helm-clean
+#############
+# Variables #
+#############
 
-CHARTS := production/helm/loki production/helm/promtail production/helm/loki-stack
+DOCKER_IMAGE_DIRS := $(patsubst %/Dockerfile,%,$(DOCKERFILES))
+IMAGE_NAMES := $(foreach dir,$(DOCKER_IMAGE_DIRS),$(patsubst %,$(IMAGE_PREFIX)%,$(shell basename $(dir))))
 
-# Boiler plate for bulding Docker containers.
-# All this must go at top of file I'm afraid.
-IMAGE_PREFIX ?= grafana/
+# Certain aspects of the build are done in containers for consistency (e.g. yacc/protobuf generation)
+# If you have the correct tools installed and you want to speed up development you can run
+# make BUILD_IN_CONTAINER=false target
+BUILD_IN_CONTAINER := true
```

Comment:
could we make this one overridable by env var `BUILD_IN_CONTAINER ?= true`

Reply:
done